parquet-hadoop/src/main/java/org/apache/parquet/crypto/ColumnDecryptionSetup.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.parquet.crypto;
+
+import org.apache.parquet.crypto.ParquetFileDecryptor.ColumnDecryptors;
+
+public class ColumnDecryptionSetup {
+  
+  private final boolean isEncrypted;
+  private final String[] columnPath;

```

Comment:
Looked at this class yesterday, for this very reason :) Unfortunately, seems to be unsuitable: the `equals`  uses `Arrays.equals`, which compares object refs, instead of obj contents. I'm using `Arrays.deepEquals`, which works fine. Should I modify `org.apache.parquet.hadoop.metadata.ColumnPath` accordingly? How will it affect the existing Parquet code? (Also, how did the existing code work till now, given this problem?).

Reply:
My bad. I saw somewhere that String array comparison requires `deepEquals`, but missed that it referred to multi-dimensional arrays. One-dimensional array is fine with a regular `equals`. I'll use the `ColumnPath`, thanks.